### ISSUE TITLE
printing and logging inconsistencies

### ISSUE THREAD

**Issue by Xabilahu** [flairNLP/flair#2654]
**Describe the bug**
There are several inconsistencies between printing with the native `print` python function and printing with `logging.getLogger('flair')`.

For example:
https://github.com/flairNLP/flair/blob/9697dc29c03ba7e5513be5a0c4f9902626917616/flair/trainers/trainer.py#L542-L543

https://github.com/flairNLP/flair/blob/9697dc29c03ba7e5513be5a0c4f9902626917616/flair/training_utils.py#L303-L304

**To Reproduce**

```
from flair.datasets import UD_ENGLISH
from flair.embeddings import WordEmbeddings
from flair.models import SequenceTagger
from flair.trainers import ModelTrainer

# 1. get the corpus
corpus = UD_ENGLISH().downsample(0.1)

# 2. what label do we want to predict?
label_type = 'upos'

# 3. make the label dictionary from the corpus
label_dict = corpus.make_label_dictionary(label_type=label_type)

# 4. initialize embeddings
embeddings = WordEmbeddings('glove')

# 5. initialize sequence tagger
tagger = SequenceTagger(hidden_size=256,
                        embeddings=embeddings,
                        tag_dictionary=label_dict,
                        tag_type=label_type,
                        use_crf=True
)

# 6. initialize trainer
trainer = ModelTrainer(tagger, corpus)

# 7. start training
trainer.train('resources/taggers/example-upos',
              learning_rate=0.1,
              mini_batch_size=32,
              max_epochs=10,
              save_model_each_k_epochs=1,
)
```

**Expected behavior**

All messages should be logged using `logger.getLogger('flair')`.

**Screenshots**

Check that the message where learning rate is reduced is not being logged.

```
2022-03-03 15:49:45,287 ----------------------------------------------------------------------------------------------------
2022-03-03 15:49:49,655 epoch 19 - iter 14/141 - loss 0.02138027 - samples/sec: 57.27 - lr: 0.100000
2022-03-03 15:49:53,916 epoch 19 - iter 28/141 - loss 0.02371238 - samples/sec: 53.10 - lr: 0.100000
2022-03-03 15:49:57,994 epoch 19 - iter 42/141 - loss 0.02502675 - samples/sec: 55.68 - lr: 0.100000
2022-03-03 15:50:02,287 epoch 19 - iter 56/141 - loss 0.02439726 - samples/sec: 52.46 - lr: 0.100000
2022-03-03 15:50:06,238 epoch 19 - iter 70/141 - loss 0.02427097 - samples/sec: 57.13 - lr: 0.100000
2022-03-03 15:50:09,520 epoch 19 - iter 84/141 - loss 0.02436758 - samples/sec: 68.99 - lr: 0.100000
2022-03-03 15:50:12,905 epoch 19 - iter 98/141 - loss 0.02479929 - samples/sec: 66.81 - lr: 0.100000
2022-03-03 15:50:17,367 epoch 19 - iter 112/141 - loss 0.02456275 - samples/sec: 54.21 - lr: 0.100000
2022-03-03 15:50:20,890 epoch 19 - iter 126/141 - loss 0.02462007 - samples/sec: 64.00 - lr: 0.100000
2022-03-03 15:50:24,996 epoch 19 - iter 140/141 - loss 0.02425031 - samples/sec: 54.84 - lr: 0.100000
2022-03-03 15:50:25,464 ----------------------------------------------------------------------------------------------------
2022-03-03 15:50:25,466 EPOCH 19 done: loss 0.0241 - lr 0.1000000
2022-03-03 15:50:30,838 DEV : loss 0.031227365136146545 - f1-score (micro avg)  0.8247
Epoch    19: reducing learning rate of group 0 to 5.0000e-02.
2022-03-03 15:50:31,099 BAD EPOCHS (no improvement): 4
2022-03-03 15:50:31,102 ----------------------------------------------------------------------------------------------------
2022-03-03 15:50:34,923 epoch 20 - iter 14/141 - loss 0.02504787 - samples/sec: 65.37 - lr: 0.050000
2022-03-03 15:50:39,327 epoch 20 - iter 28/141 - loss 0.02588841 - samples/sec: 51.29 - lr: 0.050000
2022-03-03 15:50:43,880 epoch 20 - iter 42/141 - loss 0.02613723 - samples/sec: 49.67 - lr: 0.050000
2022-03-03 15:50:47,592 epoch 20 - iter 56/141 - loss 0.02546627 - samples/sec: 60.97 - lr: 0.050000
2022-03-03 15:50:51,840 epoch 20 - iter 70/141 - loss 0.02415541 - samples/sec: 53.63 - lr: 0.050000
2022-03-03 15:50:55,439 epoch 20 - iter 84/141 - loss 0.02404081 - samples/sec: 62.85 - lr: 0.050000
2022-03-03 15:51:00,323 epoch 20 - iter 98/141 - loss 0.02341436 - samples/sec: 51.17 - lr: 0.050000
2022-03-03 15:51:04,282 epoch 20 - iter 112/141 - loss 0.02350316 - samples/sec: 56.94 - lr: 0.050000
2022-03-03 15:51:07,867 epoch 20 - iter 126/141 - loss 0.02345933 - samples/sec: 62.84 - lr: 0.050000
2022-03-03 15:51:11,720 epoch 20 - iter 140/141 - loss 0.02359901 - samples/sec: 58.54 - lr: 0.050000
2022-03-03 15:51:12,204 ----------------------------------------------------------------------------------------------------
```

**Environment (please complete the following information):**
 - OS: Arch Linux x86_64
 - Version: flair==0.10

**Additional context**

Printing with `print` in testing is ok, but it should not be like that in production ready code.

**Comment by alanakbik** [comment]
@Xabilahu thanks for spotting this, agree that the printouts should be replaced with logging statements! @Weyaaron can you take a look?

**Comment by alanakbik** [comment]
@Xabilahu thanks for the PR! 

@Weyaaron I guess this is then already taken care of ;)
